### DISCUSSION TITLE
Disappearing fluid particles

### DISCUSSION THREAD

**Discussion by carosto** [InteractiveComputerGraphics/SPlisHSPlasH#257] [Q&A] [answered]
Hello,

I currently have the issue that when running the simulation, most/all of the fluid particles disappear suddenly. 
I load the particles from a `.bgeo` file and they are placed in a jug at the start of the simulation. The jug is then rotated towards a cup to pour the particles into the cup. However, often times the particles which are still in the jug or currently pouring between jug and cup suddenly disappear from view during the pouring (all at the same time). Their positions and velocities are all set to -1 and I cannot see them anywhere in the simulation environment. 
Interestingly, the particles that are already at the bottom of the cup stay where they are.  

This is the scene file which I use:

```
{
    "Configuration": {
        "pause": true,
        "timeStepSize": 0.004167,
        "density0": 1000,
        "particleRadius": 0.006,
        "simulationMethod": 4,
        "boundaryHandlingMethod": 2,
        "gravitation": [
            0,
            -9.81,
            0
        ],
        "cflMethod": 0,
        "cflFactor": 0.5,
        "enableVTKExport": false,
        "enableRigidBodyVTKExport": false,
        "dataExportFPS": 120
    },
    "Materials": [
        {
            "id": "Fluid",
            "density0": 1000,
            "surfaceTension": 0.17,
            "surfaceTensionMethod": 2,
            "viscosity": 0.0009,
            "viscosityMethod": 1
        }
    ]
}
```
Any idea why this happens or suggestions on how to fix it?

Thanks in advance.

**Comment by janbender** [comment]
Did you try with a different boundary handling? E.g., set

"boundaryHandlingMethod": 0

Just to see if the volume maps handling is correctly configured.

**Reply by carosto** [reply]
Actually, the boundary handling method used by the simulation is already 0, despite it being set to 2 in the scene file. I manually change it in my code when initializing the simulation (`sim.setValueInt(sim.BOUNDARY_HANDLING_METHOD, 0)`). But I just checked the other two methods (1 and 2) and neither of them work, the particles just fall right through the jug. So the only working method is 0 and there the particles disappear during the simulation.

**Comment by janbender** [comment]
It is hard to say what is going on without debugging this. However, it is strange that the methods fail since we use them since several years in many challenging simulations. My guess is that there is something wrong with the boundary geometry. Did you check the particle sampling of the boundary. Does it look correct?

**Reply by carosto** [reply]
I didn't check this yet but after uninstalling and reinstalling the package multiple times, I noticed that the issue only occurs when I build it from source (following your instructions here [https://splishsplash.readthedocs.io/en/latest/build_from_source.html](url)). When I install the prebuilt package from pip the issue does not occur, at least in my tests. 
Unfortunately, I cannot simply keep using the prebuilt package, since I made some small changes (I checked and these changes are not the issue, if I build your current version here on this repository from source, the issue occurs all the same). 
Any idea why this issue only occurs when building the package from source?

**Comment by janbender** [comment, marked as answer]
That's strange. Do you use Linux or Windows? 
The prebuilt linux wheel is built without AVX support: -DUSE_AVX=Off (CMake flag)
See also here: https://github.com/InteractiveComputerGraphics/SPlisHSPlasH/blob/master/.github/workflows/build-linux.yml

Maybe there is a problem with AVX. Could you try to build the library using the CMake flag "-DUSE_AVX=Off"?

**Reply by carosto** [reply]
Building the library without AVX support seems to have solved the issue, thank you!
